src/EmailValidatorFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Egulias\EmailValidator;
+
+use Egulias\EmailValidator\EmailValidator;
+use Egulias\EmailValidator\Validation\DNSCheckValidation;
+use Egulias\EmailValidator\Validation\NoRFCWarningsValidation;
+use Egulias\EmailValidator\Validation\RFCValidation;
+
+class EmailValidatorFactory
+{
+    /** @var [] */
+    protected static array $defaultValidators = [
```

Comment:
Are you expecting or encouraging extension? This prevents it. There is also no entry point to add additional/other validators than this.

You can do it via constructor arguments with sensible defaults (is like your idea here, but with simple extension point)